wait/member.go

```diff
@@ -47,6 +50,86 @@ func (a *MemberAwaitility) WaitForUserAccount(name string, expSpec toolchainv1al
 	})
 }
 
+func (a *MemberAwaitility) WaitForNSTmplSet(name string, waitCond ...toolchainv1alpha1.Condition) error {
+	return wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		nsTmplSet := &toolchainv1alpha1.NSTemplateSet{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: a.Ns}, nsTmplSet); err != nil {
+			if errors.IsNotFound(err) {
+				a.T.Logf("waiting for availability of NSTemplateSet '%s'", name)
+				return false, nil
+			}
+			return false, err
+		}
+		if len(waitCond) != 0 && !test.ConditionsMatch(nsTmplSet.Status.Conditions, waitCond...) {
+			a.T.Logf("waiting for conditions match for NSTemplateSet '%s'", name)
+			return false, nil
+		}
+		a.T.Logf("found NSTemplateSet '%s'", name)
+		return true, nil
+	})
+}
+
+func (a *MemberAwaitility) WaitForDeletedNSTmplSet(name string) error {
+	return wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		nsTmplSet := &toolchainv1alpha1.NSTemplateSet{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: a.Ns}, nsTmplSet); err != nil {
+			if errors.IsNotFound(err) {
+				a.T.Logf("deleted NSTemplateSet '%s'", name)
+				return true, nil
+			}
+			return false, err
+		}
+		a.T.Logf("waiting for deletion of NSTemplateSet '%s'", name)
+		return false, nil
+	})
+}
+
+func (a *MemberAwaitility) GetNamespace(username, typeName string) *v1.Namespace {
+	labels := map[string]string{"owner": username, "type": typeName}
+	opts := client.MatchingLabels(labels)
+	namespaceList := &v1.NamespaceList{}
+	err := a.Client.List(context.TODO(), opts, namespaceList)
+	require.NoError(a.T, err)
+	require.True(a.T, len(namespaceList.Items) > 0)
+	return &namespaceList.Items[0]
+}
+
+func (a *MemberAwaitility) WaitForNamespace(username, typeName, revision string) error {
+	return wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		labels := map[string]string{"owner": username, "type": typeName, "revision": revision}
+		opts := client.MatchingLabels(labels)
+		namespaceList := &v1.NamespaceList{}
+		if err := a.Client.List(context.TODO(), opts, namespaceList); err != nil {
+			return false, err
+		}
+
+		if len(namespaceList.Items) < 1 {
+			a.T.Logf("waiting for availability of Namespace type '%s' with revision '%s'", typeName, revision)
+			return false, nil
+		}
+		a.T.Logf("found Namespace type '%s' with revision '%s'", typeName, revision)
```

Comment:
I believe that we could also check that there is only one namespace found
```suggestion
		require.Len(a.T, namespaceList.Items, 1, "there should be only one Namespace found")
		a.T.Logf("found Namespace type '%s' with revision '%s'", typeName, revision)
```